File: OP2/lab_5/lab_5_python/TRealNumber.py
from TNumber import TNumber


class TRealNumber(TNumber):
    def __init__(self, num):
        super().__init__(int(str(num).partition('.')[0]))
        self.__frac = int(str(num).partition('.')[2])

    def __repr__(self):
        return str(self._number) + '.' + str(self.__frac)

    def get_number(self):
        return float(str(self._number)+'.'+str(self.__frac))

    def find_last_digit(self):
        return int(str(self.__frac)[-1])

File: OP2/lab_5/lab_5_c++/lab_5/TRealNumber.cpp
#include "TRealNumber.h"
#include <string>

TRealNumber::TRealNumber(float num)
{
	std::string str = std::to_string(num);
	int p = str.find('.');
	number = std::stoi(str.substr(0,p));
	frac = std::stoi(str.substr(p + 1));
}

std::string TRealNumber::get_str_number()
{
	return std::to_string(number) + '.' + std::to_string(frac);
}

int TRealNumber::find_last_digit()
{
	std::string str = std::to_string(frac);
	return str[str.length() - 1] - '0';
}
